### PR TITLE
feat(fetch): import/export storageState

### PR DESCRIPTION
**Pull request by yury-s**
#5999

### CODE REVIEW

File: docs/src/api/class-playwright.md [hunk 1]
```diff
@@ -112,6 +112,27 @@ When using [`method: FetchRequest.get`], [`method: FetchRequest.post`], [`method
...
+    - `sameSite` <[SameSiteAttribute]<"Strict"|"Lax"|"None">>
+  - `origins` <[Array]<[Object]>>
+    - `origin` <[string]>
+    - `localStorage` <[Array]<[Object]>>
+      - `name` <[string]>
+      - `value` <[string]>
+
+Populates context with given storage state. This option can be used to initialize context with logged-in information
+obtained via [`method: BrowserContext.storageState`] or [`method: FetchRequest.storageState`]. Either a path to the
+file with saved storage, or the value returned by one of the `storageStgate` methods.
```
Comment (dgozman) [inline docs/src/api/class-playwright.md:134]: - typo: storageStgate
- Let's link directly to both `storageState` methods instead? As a user, I have no idea what you are talking about :)

File: src/client/types.ts [hunk 2]
```diff
@@ -118,3 +118,22 @@ export type SelectorEngine = {
...
+  ignoreHTTPSErrors?: boolean;
+  proxy?: {
+    server: string;
+    bypass?: string;
+    username?: string;
+    password?: string;
+  };
+  timeout?: number;
+  userAgent?: string;
+  storageState?: string | channels.BrowserNewContextOptions['storageState'];
```
Comment (dgozman) [inline src/client/types.ts:138]: Should this be `channels.PlaywrightNewRequestOptions` instead? Also, there is `StorageState` defined above in this file.

File: src/client/types.ts [hunk 3]
```diff
@@ -118,3 +118,22 @@ export type SelectorEngine = {
 
 export type RemoteAddr = channels.RemoteAddr;
 export type SecurityDetails = channels.SecurityDetails;
+
+export type NewRequestOptions = {
```
Comment (dgozman) [inline src/client/types.ts:122]: Why not the following?
```js
export type NewRequestOptions = Omit<channels.PlaywrightNewRequestOptions, 'extraHTTPHeaders' | 'storageState'> & {
  extraHTTPHeaders?: Headers,
  storageState?: string | channels.PlaywrightNewRequestOptions['storageState'],
};
```

**Review by dgozman — APPROVED** [review]

File: src/client/types.ts (hunk 2 above)
Comment (yury-s) [reply src/client/types.ts:138]: Changed to StorageState.

File: src/client/types.ts (hunk 3 above)
Comment (yury-s) [reply src/client/types.ts:122]: Yeah, afterthoughts... changed.

**Pull request merged**